### ISSUE TITLE
Support XML elements in TypeScript <ELEMENT ... value=$VALUE  .../>

### ISSUE THREAD

**Issue by LewisArdern** [semgrep/semgrep#5115]
**Describe the bug**
XML elements aren't supported by the Generic-to-IL translation, in TypeScript semgrep should be able to capture:
``` 
(<SourceLink ... href=$HREF .../>)
 (<SourceLink ... href={$HREF} .../>)
```
It currently captures 
```
(<SourceLink ... href={...} .../>)
```
But this is not great for allowing people to really dig deep into TypeScript values.

**To Reproduce**
Bug: 
https://semgrep.dev/s/K9l5
Workaround
https://semgrep.dev/s/0PQj

**Expected behavior**
Works: https://semgrep.dev/s/K9l5

**What is the priority of the bug to you?**

- [x] P2: regular bug that should get fixed

**Use case**
Enable typescript to support better inspection and build better rules by customers.

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-1224/support-xml-elements-in-typescript-element-value=dollarvalue. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by aryx** [comment]
@IagoAbal is this fixed now?

**Comment by IagoAbal** [comment]
Ooops, sorry, due to a typo in PR #5434 it closed #4115 instead of this issue.
